### ISSUE TITLE
Property Naming Conventions

### ISSUE THREAD

**Issue by goneall** [spdx/spdx-spec#198]
This issue is spun off from the discussion in PR #197 related to property naming conventions for the JSON property names.

We currently have defined property names and conventions for RDF/XML and Tag/Value.

With the introduction of YAML, JSON and XML formats in 2.2, there is a question on what the naming conventions should be for those formats.

We can break this down into a couple of questions:
- Should we have a common property naming convention between all formats?
- If so, what should that format be?
- If not, what is the naming conventions be for each of the formats?
- If not, how do we make it easy for tool providers to support multiple formats?

**Comment by goneall** [comment]
From @pombredanne in PR #197 

I would also rather prefer to use `file_dependencies` rather than `fileDependencies`. Camel case is harder and slower to read. For instance:
`aRatherLongSymbolName` vs `a_rather_long_symbol_name`

See also https://whatheco.de/2013/02/16/camelcase-vs-underscores-revisited/

**Comment by goneall** [comment]
From @goneall in PR #197 
> I would also rather prefer to use file_dependencies rather than fileDependencies.

I would prefer to keep the property names consistent with the other formats for the 2.2 release.  It would be a rather significant change to the tooling to translate between the different conventions.

We could pick up the discussion in 3.0.  I would suggest we first discuss/agree on whether it is important for the names to be consistent between the various formats (e.g. XML, YAML, JSON, RDF/XML, Tag/Value) and how compatible we would want to be with previous releases.

**Comment by goneall** [comment]
From @davaya in PR#197

> IMHO list of things should be plural. This is a sane convention and proper English.
> I would also rather prefer to use `file_dependencies` rather than `fileDependencies`. Camel case is harder and slower to read. For instance:
> `aRatherLongSymbolName` vs `a_rather_long_symbol_name`
> 
> See also https://whatheco.de/2013/02/16/camelcase-vs-underscores-revisited/

Speaking as an interested outsider, it is useful to articulate naming conventions and then validate names in a specification against a regex to ensure that the conventions are followed consistently.   A protocol I'm currently working on uses lower_case_with_underscores for property names, and Camel-Case-With-Dashes for schema type names, so +1 for using lower_underscore or lower-dash property names.

+1 also for using plurals. Plurals shouldn't necessarily be generated by tooling; the human that creates a type that is an array of "Party" should be able to explicitly name that type "Parties", and the property names used in instances of those types are always assigned by the human, and need not have any relationship to the type name, e.g., IP-Pair = Array(src_addr: IP-Address, dst_addr: IP-Address).

**Comment by goneall** [comment]
If we do go down the path of different names, we could create a JSON file that maps the names for the different formats.  Something like:
```
[
    {"json":"relationhip_type",
     "rdf":"spdx:relationshipType",
     "xml":"RelationshipType",
     "yaml":relationship_type",
     "tagvalue":""
     }, ...
]
```

We could even include the description and autogenerate some of the schemas and documentation.

**Comment by goneall** [comment]
The sig-security-sbom workgroup is taking a JSON configuration file oriented approach.  The configuration files are located at https://github.com/cdfoundation/sig-security-sbom/tree/master/modeling/model_configuration

Much of this work was done by @castresearchlabs

**Comment by goneall** [comment]
I believe this has been decided and resolved in 3.0

Closing issue.
